### PR TITLE
feat(docs): auto-import interfaces, types, enum declarations in the docs from TS sources

### PR DESCRIPTION
**Pull request by AlCalzone**
With this PR, we have the ability to import the actual definition of types, interfaces and enums into documentation. To do so, I've introduced a comment syntax that is similar to TypeScripts imports:

```md
<!-- #import Firmware from "zwave-js" -->
```
will be converted to
~~~md
<!-- #import Firmware from "zwave-js" -->

```ts
interface Firmware {
    data: Buffer;
    firmwareTarget?: number;
}
```
~~~
for example. There are a few kinks to work out, but the basic functionality is there.

**Next steps:**

- [x] review the generated files if we like the changes
- [x] add the option to remove comments (e.g. if we have better comments in the docs)
- [ ] use this in a github action that checks if the docs can be built
- [ ] use this action to detect changes and create a PR if changes are detected

### CODE REVIEW

File: docs/api/driver.md [hunk 1]
```diff
@@ -244,27 +248,32 @@ interface FileSystem {
...
 	supportCheck?: boolean;
-	/** Whether the driver should update the node status to asleep or dead when a transaction is not acknowledged (repeatedly). Setting this to false will cause the simply transaction to be rejected on failure. Default: true */
+	/**
+	 * Whether the driver should update the node status to asleep or dead when a transaction is not acknowledged (repeatedly).
+	 * Setting this to false will cause the simply transaction to be rejected on failure.
+	 * Default: true
+	 */
 	changeNodeStatusOnMissingACK?: boolean;
-	/** Sets the number of milliseconds after which a message expires. When the expiration timer elapses, the promise is rejected with the error code `Controller_MessageExpired`. */
+	/** Sets the number of milliseconds after which a transaction expires. When the expiration timer elapses, the transaction promise is rejected. */
```
Comment (AlCalzone) [inline docs/api/driver.md]: 👎

File: docs/api/driver.md [hunk 2]
```diff
@@ -244,27 +248,32 @@ interface FileSystem {
...
 	expire?: number;
+	/** Internal information used to identify or mark this transaction */
+	tag?: any;
```
Comment (AlCalzone) [inline docs/api/driver.md]: @internal

File: docs/api/driver.md [hunk 3]
```diff
@@ -275,26 +284,30 @@ enum MessagePriority {
 	// Our handshake requests must be prioritized over all other messages
 	PreTransmitHandshake = 1,
 	// Controller commands usually finish quickly and should be preferred over node queries
-	Controller = 2,
+	Controller,
```
Comment (AlCalzone) [inline docs/api/driver.md:285]: Not sure, nobody should use the numbers, so there's no need to document them.

File: docs/api/node.md [hunk 4]
```diff
@@ -205,25 +209,30 @@ enum InterviewStage {
 	ProtocolInfo,
 	/** The node has been queried for supported and controlled command classes */
 	NodeInfo,
+
```
Comment (AlCalzone) [inline docs/api/node.md:212]: This comes from removed single-line comments. Not pretty, but I see no simple way around it.

File: docs/api/valueid.md [hunk 5]
```diff
@@ -68,14 +77,22 @@ interface ValueMetadataAny extends ValueMetadataBase {
...
 	max?: number;
+	/** When only certain values between min and max are allowed, this determines the step size */
 	steps?: number;
+	/** The default value */
 	default?: number;
+	/** Speaking names for numeric values */
 	states?: Record<number, string>;
+	/** An optional unit for numeric values */
 	unit?: string;
 }
```
Comment (AlCalzone) [inline docs/api/valueid.md]: 👎

File: docs/api/valueid.md [hunk 6]
```diff
@@ -89,9 +106,12 @@ interface ValueMetadataNumeric extends ValueMetadataBase {
 
 #### boolean
 
+<!-- #import ValueMetadataBoolean from "zwave-js" -->
+
 ```ts
 interface ValueMetadataBoolean extends ValueMetadataBase {
 	type: "boolean";
+	/** The default value */
```
Comment (AlCalzone) [inline docs/api/valueid.md]: 👎

File: docs/api/valueid.md [hunk 7]
```diff
@@ -100,11 +120,16 @@ interface ValueMetadataBoolean extends ValueMetadataBase {
...
+<!-- #import ValueMetadataString from "zwave-js" -->
+
 ```ts
 interface ValueMetadataString extends ValueMetadataBase {
 	type: "string";
+	/** The minimum length this string must have (optional) */
 	minLength?: number;
+	/** The maximum length this string may have (optional) */
 	maxLength?: number;
+	/** The default value */
```
Comment (AlCalzone) [inline docs/api/valueid.md]: :-1:
but the description below should be synchronized with this.

File: docs/api/valueid.md [hunk 8]
```diff
@@ -139,11 +168,13 @@ type BinarySensorValueMetadata = ValueMetadata & {
...
 #### Indicator CC
 
+<!-- #import IndicatorMetadata from "zwave-js" -->
+
 ```ts
 type IndicatorMetadata = ValueMetadata & {
 	ccSpecific: {
 		indicatorId: number;
-		propertyId?: number; // only present on V2+ indicators
+		propertyId?: number;
```
Comment (AlCalzone) [inline docs/api/valueid.md]: :-1:

File: maintenance/generateTypedDocs.ts [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+	module: string;
+	symbol: string;
+	import: string;
+	options: {
+		comments?: boolean;
+		jsdoc?: boolean;
+	};
+}
+
+const importRegex = /(?<import><!-- #import (?<symbol>.*?) from "(?<module>.*?)"(?: with (?<options>[\w\-, ]*?))? -->)(?:[\s\r\n]*```ts[\r\n]*(?<source>(.|\n)*?)```)?/g;
```
Comment (robertsLando) [inline maintenance/generateTypedDocs.ts:132]: @AlCalzone This regex is pure madness :rofl:

**Review by robertsLando — APPROVED** [review]
Everything seems good here. I never used `ts-morph` so cannot say much about the docs generation methods

**Comment by AlCalzone** [comment]
> I never used `ts-morph` so cannot say much about the docs generation methods

It is basically a tool to manipulate the abstract syntax tree (AST) of TypeScript. I just found it today, but it nicely abstracts away the somewhat cumbersome methods the TS API has to offer.
A good example is the one-liner `toggleModifiers` instead of my (incomplete) `updateModifiers` implementation in this commit:
https://github.com/zwave-js/node-zwave-js/pull/1148/commits/c33549169aa27603e2c7876a7f5e5aff1f499be6#

File: maintenance/generateTypedDocs.ts (hunk 9 above)
Comment (AlCalzone) [reply maintenance/generateTypedDocs.ts:132]: I've seen worse

**Pull request merged**